### ISSUE TITLE
Detect computed properties in constructor

### ISSUE THREAD

**Issue by aditya-kappagantula** [Intellicode/eslint-plugin-react-native#187]
```JavaScript
  // Component WelcomeBar
  import React, { Component } from 'react';
  import { Text, View, Image } from 'react-native';
  import styles from './styles';

  export default class WelcomeBar extends Component {
    constructor(props) {
      super(props);
      // Code Smell
      this.state = {fullName: this.props.firstName + this.props.lastName}
    }
    render() {
      // This will not change on subsequent renders
      return (<Text>{this.state.fullName}</Text>);
    }
  }


  // Component WelcomePage
  import React, { Component } from 'react';
  import { Text, View, Image } from 'react-native';
  import styles from './styles';
  export default class WelcomePage extends Component {
    someMethod() {
      this.setState({lastName: 'Modified'});
    }

    render() {
      return (
        <View>
          <WelcomeBar isUserLoggedIn={true} firstName=this.state.firstName lastName=this.state.lastName />
          <WelcomeMessage />
        </View>
      );
    }
  }
```
In the above example, `WelcomeBar` is re-rendered by `WelcomePage` when `someMethod` is executed. However it doesn't update the fullName.
This could have been avoided by listening to the deprecated `componentWillReceiveProps` or the `getDerivedStateFromProps`. However, this kind of code is smell to be in the first place. Can this be detected in es-lint please?

**Comment by Intellicode** [comment]
Hi @aditya-kappagantula, this may be more suitable to do in https://github.com/yannickcr/eslint-plugin-react/issues
